src/main/java/io/temporal/samples/interceptor/SimpleCountWorkflowInboundCallsInterceptor.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.interceptor;
+
+import io.temporal.common.interceptors.WorkflowInboundCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowInboundCallsInterceptorBase;
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
+import io.temporal.samples.interceptor.collector.CountCollector;
+
+public class SimpleCountWorkflowInboundCallsInterceptor
+    extends WorkflowInboundCallsInterceptorBase {
+
+  public SimpleCountWorkflowInboundCallsInterceptor(WorkflowInboundCallsInterceptor next) {
+    super(next);
+  }
+
+  @Override
+  public void init(WorkflowOutboundCallsInterceptor outboundCalls) {
+    super.init(new SimpleCountWorkflowOutboundCallsInterceptor(outboundCalls));
+  }
+
+  @Override
+  public WorkflowOutput execute(WorkflowInput input) {
+
```

Comment:
Did we miss counting here?

Reply:
Not really. If so it would be double count. The count for this is done in SimpleCountWorkflowOutboundCallsInterceptor

Reply:
1. In that case what's the point in overriding a method just to make `return super.execute(input);`? We can just remove it.
2. I think it's a good idea to move the counting here for a couple of reasons:
-  you now are handling the counting of executions and signals/queries inconsistently. You count one in outbound interceptors while another in inbound interceptors. There is no reason for that, it looks just like an inconsistent implementation.
- It looks like you miss counting of the `execute` of the main root workflow. Because this would be the place to do that.

Even if you leave the current implementation - at least it should be a javadoc here about what's going on and why we count signals and queries here but don't count executions. The code itself is not self-documenting enough in this case.

Reply:
In WorkflowOutboundCallsInterceptorBase you can call Workflow.getInfo() which I got from your interceptor impl actually :) Can try to move to this one no problemo